### DISCUSSION TITLE
How should I use TEXT_FIELD_TYPE ?

### DISCUSSION THREAD

**Discussion by ajoslin103** [lookfirst/mui-rff#663] [Q&A]
I want to type a field to only accept numbers or text - I can't understand how to do this...

I import the library and types

      // https://github.com/lookfirst/mui-rff
      import { TextField, TEXT_FIELD_TYPE, TYPE_TEXT, TYPE_NUMBER } from "mui-rff";

And I immediately get type errors

      TS2305: Module '"mui-rff"' has no exported member 'TEXT_FIELD_TYPE'.
      TS2305: Module '"mui-rff"' has no exported member 'TYPE_TEXT'.
      TS2305: Module '"mui-rff"' has no exported member 'TYPE_NUMBER'.

What am I doing wrong?

**Comment by lookfirst** [comment]
Can I recommend something to you? Don't do that. It is a UX usability nightmare. The number type for an input field is not implemented well or consistently across browsers and platforms.

**Reply by ajoslin103** [reply]
That's good advice with a sound reason.

How would you suggest I implement numbers only ?

Note: this is for use in an Electron App -- so I think I'm safe from cross-browser problems, and maybe from cross-platform issues. So far it seems good, but I'll have to wait until this upcoming week to see how Windows users fare.

**Comment by lookfirst** [comment]
Even in electron, "well" still applies.

Use yup for validation and show an error.
